### PR TITLE
Milestone3 game logic

### PR DESCRIPTION
**Pull request by Ahmed-Benj**
Hello,

In this milestone I have added a the game logic to the lib folder and included the methods :

1. won
2. full board
3. position taken check
4. WINCOMBINATIONS

Could you please review my project so far ?

Kind regards,

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,89 +1,85 @@
 #!/usr/bin/env ruby
-##rubocop:disable all
+#rubocop:disable all
 
-def querydata(dispmsg)
-  puts dispmsg
-  gets.chomp.strip
-end
+require '../lib/tic_tac_toe'
+
```
Comment (elisha2kyakpo1) [inline bin/main.rb]: Use require_relative instead of require

File: bin/main.rb [hunk 2]
```diff
@@ -1,89 +1,85 @@
...
-def display_board(board, symbol, position)
-  board = board.map { |x| x == position ? symbol : x }
-  puts '+---+---+---+'
-  puts "| #{board[0]} | #{board[1]} | #{board[2]} |"
-  puts '+---+---+---+'
-  puts "| #{board[3]} | #{board[4]} | #{board[5]} |"
-  puts '+---+---+---+'
-  puts "| #{board[6]} | #{board[7]} | #{board[8]} |"
-  puts "+---+---+---+ \n\n"
-  board
```
Comment (elisha2kyakpo1) [inline bin/main.rb]: No puts outside main.rb file

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
# Status: Changes required💪✔🎉
Hey @Ahmed-Benj, you have done a good job so far. However, there is room for improvement.

- [x] Fix the rubocop errors
- [x] I have seen useless files .vscode, .gitkeep in your project, please remove/delete them.
- [x] Why do you have two files in bin folder? just maintain one with the latest code.
- [x] No puts/print and gets.chomp outside main.rb file.
- [x] Use require_relate instead of require.
- [x] The code of the project should be logically divided between classes. Create at least two classes in lib folder to accommodate your lib folder code ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

<hr>
[OPTIONAL]

- [x] The game accepts numbers for a player name. You can make validations for this as well.

<hr>
You can reach me on slack by searching my name as Elisha kyakopo in case you have a question.

Happy coding💪👈

File: bin/main.rb (hunk 2 above)
Comment (Ahmed-Benj) [reply bin/main.rb]: > No puts outside main.rb file

Fixed

File: bin/main.rb (hunk 1 above)
Comment (Ahmed-Benj) [reply bin/main.rb]: > Use require_relative instead of require

Fixed

**Comment by Ahmed-Benj** [comment]
Hello @elisha2kyakpo1 ,

Thank you for these comments.

I have made the requested changes, could you please re-review my project.

Kind regards,
Ahmed

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
# Status: Changes required👈
Hey @Ahmed-Benj, I'm happy to see almost all the changes requested being implemented, just one issue has prevented this project from being approved.

- [x] The code of the project should be logically divided between classes. Create at least two classes in lib folder to accommodate your lib folder code (screenshot from Pathwright)

That's it, make the change and request for a review.

Once again, you can reach me on slack by searching Elisha kyakopo, in case you have a question.

Cheers🎉✔💪

**Comment by Ahmed-Benj** [comment]
I couldn't find a way to add another class in the lib folder, I don't know why I would need that in my solution, 

could you please approve my project ?

Kind regards,

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: :clap: 
---------
Hello @Ahmed-Benj  Great job you did on the required changes :clap: :muscle: 
The project is working really nice :+1:  good job :1st_place_medal: 
--------
Now you can merge this branch :100: :smile: 
HAPPY CODING!! AND GOOD LUCK WITH THE COMMING MILESTONE :crossed_fingers: :four_leaf_clover:

tenor

**Comment by Ahmed-Benj** [comment]
> ## STATUS: APPROVED 🟢 👏
> ## Hello @Ahmed-Benj Great job you did on the required changes 👏 💪
> The project is working really nice 👍 good job 🥇
> Now you can merge this branch 💯 😄
> HAPPY CODING!! AND GOOD LUCK WITH THE COMMING MILESTONE 🤞 🍀
> 
> tenor

Hello Brian, thanks for the review!

**Pull request merged**